airflow/jobs/local_task_job_runner.py

```diff
@@ -303,6 +306,14 @@ def heartbeat_callback(self, session: Session = NEW_SESSION) -> None:
                 if dagrun_timeout and execution_time > dagrun_timeout:
                     self.log.warning("DagRun timed out after %s.", execution_time)
 
+            # If process still runs after being marked as success, let it run until configured overtime
+            # if there are configured listeners
+            if (
+                ti.state == TaskInstanceState.SUCCESS
+                and self._overtime < conf.getint("core", "task_listener_timeout")
+                and get_listener_manager().has_listeners
```

Comment:
Do we want to limit it to only case where we have listeners? I think very similar scenario might happen when there is a "mini-scheduler" running after task completes. I **think** that case would also benefit from eager heartbeat killing such task. Maybe then we should turn it into a generic "task_timeout" not "task_listener_timeout" and kill two birds with the same stone?

Reply:
Ok - I've just removed the check for listeners and renamed it as `task_success_overtime` - should cover that needs.